File: NZSLDict/Classes/ViewControllerPad.m
//
//  ViewControllerPad.m
//  NZSL Dict
//
//  Created by Greg Hewgill on 27/04/13.
//
//

#import "ViewControllerPad.h"

#import "DetailViewController.h"
#import "DiagramViewController.h"
#import "SearchViewController.h"
#import "VideoViewController.h"
#import "HistoryViewController.h"
#import "AboutViewController.h"

@interface ViewControllerPad ()

@end

@implementation ViewControllerPad {
    SearchViewController *searchController;
    HistoryViewController *historyController;
    DetailViewController *detailViewController;
    UIViewController *diagramController;
    UIViewController *videoController;
}

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
        UITabBarController *tabbar = [[UITabBarController alloc] init];
        tabbar.viewControllers = @[
            searchController = [[SearchViewController alloc] init],
            historyController = [[HistoryViewController alloc] init],
        ];
        self.viewControllers = @[
            tabbar,
            detailViewController = [[DetailViewController alloc] init],
        ];
        self.delegate = detailViewController;
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view.
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)toInterfaceOrientation
{
    return YES;
}

@end

File: NZSLDict/Classes/ViewControllerPad.swift
import Foundation
import UIKit

class ViewControllerPad: UISplitViewController {
    var searchController: SearchViewController!
    var historyController: HistoryViewController!
    var detailController: DetailViewController!

    // These were in the ObjC version but I don't think they are used now
    // var diagramController: UIViewController!
    // var videoController: UIViewController!

    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)

        searchController = SearchViewController()
        historyController = HistoryViewController()
        detailController = DetailViewController()

        let tabbar = UITabBarController()
        tabbar.viewControllers = [searchController, historyController]

        self.viewControllers = [tabbar, detailController]
        self.preferredDisplayMode = .allVisible
        self.delegate = detailController
    }
    
    override func viewDidLayoutSubviews() {
        let kMasterViewWidth: CGFloat = 320.0

        let masterViewController = viewControllers[0]
        let detailViewController = viewControllers[1]

        if detailViewController.view.frame.origin.x > 0.0 {
            // Adjust the width of the master view
            var masterViewFrame = masterViewController.view.frame
            let deltaX = masterViewFrame.size.width - kMasterViewWidth
            masterViewFrame.size.width -= deltaX
            masterViewController.view.frame = masterViewFrame

            // Adjust the width of the detail view
            var detailViewFrame = detailViewController.view.frame
            detailViewFrame.origin.x -= deltaX
            detailViewFrame.size.width += deltaX
            detailViewController.view.frame = detailViewFrame

            masterViewController.view.setNeedsLayout()
            detailViewController.view.setNeedsLayout()
        }
    }

    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }

    override var shouldAutorotate : Bool {
        return true
    }
}
